pkg/generate/code/set_resource.go

```diff
@@ -119,13 +119,46 @@ func SetResource(
 	}
 
 	// We might be in a "wrapper" shape. Unwrap it to find the real object
-	// representation for the CRD's createOp. If there is a single member
-	// shape and that member shape is a structure, unwrap it.
-	if outputShape.UsedAsOutput && len(outputShape.MemberRefs) == 1 {
-		for memberName, memberRef := range outputShape.MemberRefs {
-			if memberRef.Shape.Type == "structure" {
-				sourceVarName += "." + memberName
-				outputShape = memberRef.Shape
+	// representation for the CRD's createOp/DescribeOP.
+
+	// Use the wrapper field path if it's given in the ack-generate config file.
+	wrapperFieldPath := r.SetOutputWrapperFieldPath(op)
```

Comment:
I understand this field comes from the `set_output_wrapper_field_path` JSON key, but it's pretty confusing to see a getter with the name `Set*`.

Reply:
Yes, i agree it's very confusing.. I'm not sure if `GetSetOutputWrapperFieldPath` is a better option.

Reply:
We also have a similar function is crd.go: `SetOutputCustomMethodName`

Reply:
I understand using `set_*` for the JSON key, but the getter for this value could just be `GetOutputWrapperFieldPath`? 

```go
	OutputWrapperFieldPath string `json:"set_output_wrapper_field_path,omitempty"`
```

then:

```go
 	wrapperFieldPath := r.OutputWrapperFieldPath(op) 
```

Perhaps?

Reply:
++ in this case I prefer to remove the `set_` from the JSON Key as well - just to keep fields names and json tags synced

Reply:
Renamed to `OutputWrapperFieldPath`